crates/testing/src/state_types.rs

```diff
@@ -60,36 +60,32 @@ impl State for TestState {
 
     type Time = ViewNumber;
 
-    fn validate_block(&self, _block_header: &Self::BlockHeader, view_number: &Self::Time) -> bool {
-        if view_number == &ViewNumber::genesis() {
-            &self.view_number == view_number
-        } else {
-            self.view_number < *view_number
-        }
-    }
-
-    fn initialize() -> Self {
-        let mut state = Self::default();
-        state.block_height += 1;
-        state
-    }
-
-    fn append(
+    fn validate_and_apply_header(
         &self,
-        block_header: &Self::BlockHeader,
+        _proposed_header: &Self::BlockHeader,
+        _parent_header: &Self::BlockHeader,
         view_number: &Self::Time,
     ) -> Result<Self, Self::Error> {
-        if !self.validate_block(block_header, view_number) {
+        if view_number == &ViewNumber::genesis() {
+            if &self.view_number != view_number {
+                return Err(BlockError::InvalidBlockHeader);
+            }
+        } else if self.view_number >= *view_number {
             return Err(BlockError::InvalidBlockHeader);
         }
-
         Ok(TestState {
             block_height: self.block_height + 1,
             view_number: *view_number,
             prev_state_commitment: self.commit(),
         })
     }
 
+    fn initialize() -> Self {
+        let mut state = Self::default();
+        state.block_height += 1;
```

Comment:
This also seems problematic. We're basically taking a wild guess about the block height if we missed some blocks, but if we're wrong we won't be able to validate or vote on anything. This should take a header and get the correct block height using `BlockHeader::block_number()`

Reply:
As mentioned in the other comment, I'll remove this function for now since it's unused.